ipu-speed-advantage/benchmarking_gnn_on_ipus_review.ipynb

```diff
@@ -0,0 +1,867 @@
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4a466dc3",
+   "metadata": {},
+   "source": [
+    "### Running on Paperspace\n",
+    "\n",
+    "The Paperspace environment lets you run this notebook with minimal effort. To improve your experience we preload datasets and pre-install packages, this can take a few minutes. If you experience errors immediately after starting a session, please try restarting the kernel before contacting support. If a problem persists or you want to give us feedback on the content of this notebook, please reach out to us through our community of developers using our [slack channel](https://www.graphcore.ai/join-community) or raise a [GitHub issue](https://github.com/graphcore/examples).\n",
+    "\n",
+    "To set up the requirements for running the session, simply run the following line:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6d15f513",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.0.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%pip install -q -r requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e51b6334",
+   "metadata": {},
+   "source": [
+    "And for compatibility with the Paperspace environment variables we will do the following:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "62b6533f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import poptorch\n",
+    "\n",
+    "poptorch.setLogLevel(\"ERR\")\n",
+    "executable_cache_dir = (\n",
+    "    os.getenv(\"POPLAR_EXECUTABLE_CACHE_DIR\", \"/tmp/exe_cache/\") + \"/pyg-benchmark\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bc6f84ad",
+   "metadata": {},
+   "source": [
+    "Now we are ready to benchmark gather and scatter operations on the IPU."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "737edde2",
+   "metadata": {},
+   "source": [
+    "## Setting up the benchmarks"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9542d774",
+   "metadata": {},
+   "source": [
+    "In this section we will discuss how we will run the benchmarks in this notebook on the IPU.\n",
+    "\n",
+    "First we create a simple module which runs a scatter operation. \n",
+    "\n",
+    "This module does the following:\n",
+    " * Creates random input node and edge tensors of sizes `[num_inputs, num_features]` and `[num_outputs, num_inputs]` respectively,\n",
```

Comment:
Is it one input node or many? 

If one it should be "Creates a random input node".
If many, it should be "Creates random input nodes".